mirage/views/addons.ts

```diff
@@ -315,13 +315,89 @@ export function createAuthorizedComputingAccount(this: HandlerContext, schema: S
     }
 }
 
+export function updateAuthorizedStorageAccount(this: HandlerContext, schema: Schema) {
+    const attrs  = this.normalizedRequestAttrs(
+        'authorized-storage-account',
+    ) as NormalizedRequestAttrs<MirageAuthorizedStorageAccount>;
+    const externalService = schema.externalStorageServices
+        .find(attrs.storageProviderId) as ModelInstance<ExternalStorageServiceModel>;
+    try {
+        const authorizedAccount = schema.authorizedStorageAccounts.find(attrs.id);
+        let authorizedAttrs = attrs;
+        if (attrs.credentials || attrs.initiateOauth) {
+            authorizedAttrs = prepareAuthorizedAccountAttrs(
+                attrs, externalService,
+            ) as NormalizedRequestAttrs<MirageAuthorizedStorageAccount>;
+        }
+        authorizedAccount.update({
+            ...authorizedAttrs,
+        });
+        return authorizedAccount;
+    } catch (e) {
+        return new Response(403, {}, {
+            errors: [{ detail: e.message }],
+        });
+    }
```

Comment:
So this try will catch if there's a permissions issue?

Reply:
Yea, in `prepareAuthorizedAccountAttrs`, we call `fakeCheckCredentials` that will throw an error if the necessary credentials fields aren't truthy